ui/chat/src/discussion.ts

```diff
@@ -84,11 +84,18 @@ function renderInput(ctrl: Ctrl): VNode | undefined {
 let mouchListener: EventListener;
 
 const setupHooks = (ctrl: Ctrl, chatEl: HTMLElement) => {
-  chatEl.addEventListener('keypress',
+  chatEl.addEventListener('keydown',
     (e: KeyboardEvent) => setTimeout(() => {
       const el = e.target as HTMLInputElement,
         txt = el.value,
         pub = ctrl.opts.public;
+      if(sessionStorage.getItem("txt") != null && sessionStorage.getItem("txt") != "") {
+        el.value = sessionStorage.getItem("txt")!
+        sessionStorage.removeItem("txt")
+      }
+      window.addEventListener('beforeunload', () => {
```

Comment:
you definitely don't want to call `addEventListener` whenever a `keydown` is triggered.